src/sst.rs

```diff
@@ -696,14 +709,51 @@ mod tests {
                 "first key in sst info should be correct"
             );
         }
+
+        #[allow(unused)]
+        async fn test_compression_using_sst_info(compression: CompressionCodec) {
+            let root_path = Path::from("");
+            let object_store: Arc<dyn ObjectStore> = Arc::new(InMemory::new());
+            let format = SsTableFormat::new(4096, 0, Some(compression));
+            let table_store = TableStore::new(object_store.clone(), format, root_path.clone());
+            let mut builder = table_store.table_builder();
+            builder.add(b"key1", Some(b"value1")).unwrap();
+            builder.add(b"key2", Some(b"value2")).unwrap();
+            let encoded = builder.build().unwrap();
+            let encoded_info = encoded.info.clone();
+            table_store
+                .write_sst(&SsTableId::Wal(0), encoded)
+                .await
+                .unwrap();
+
+            // Decompression is independent of TableFormat. It uses the CompressionFormat from SSTable Info to decompress sst.
+            let format = SsTableFormat::new(4096, 0, Some(CompressionCodec::Snappy));
```

Comment:
For the Snappy compression codec, this will use the same compression for both encode and decode SsTableFormats. Can you do two parameters for this test so you can change the second format for snappy.